File: request.go
package main

import (
	"encoding/json"
	"fmt"
	"log"
	"net/http"
)

func main() {
	// START
	url := "https://httpbin.org/ip"

	resp, err := http.Get(url)
	if err != nil {
		log.Fatalf("error: can't get %q - %s", url, err)
	}
	defer resp.Body.Close()

	dec := json.NewDecoder(resp.Body)
	var reply struct {
		Origin string `json:"origin"`
	}
	if err := dec.Decode(&reply); err != nil {
		log.Fatalf("error: can't decode reply - %s", err)
	}
	fmt.Println(reply.Origin)
	// END
}

File: request.py
from urllib.request import urlopen, HTTPError
import json

# START
url = 'https://httpbin.org/ip'
try:
    with urlopen(url) as fp:
        reply = json.load(fp)
except HTTPError as err:
    msg = 'error: cannot get {!r} - {}'.format(url, err)
    raise SystemExit(msg)
except ValueError as err:
    msg = 'error: cannot decode reply - {}'.format(err)
    raise SystemExit(msg)

print(reply['origin'])
# END
